doxygen/contributor_help_pages/require_meta.md

```diff
@@ -49,135 +285,141 @@ class a_class<T, require_std_vector_st<is_var, T>> {
+    ```
+
+There variant of the requires that places the `vt` or `st` before the
+type trait name only checks the @ref value_type or @ref `scalar_type`
+of `T` without testing `T`.
 
-```cpp
- template <typename Mat1, typename Mat2,
-  require_all_eigen_vt<is_arithmetic, Mat1, Mat2>* = nullptr>
- inline auto a_func(Mat1&& m1, Mat2&& m2) {
-   check_not_nan(m1);
-   check_not_nan(m2);
-   return another_func(std::forward<Mat1>(m1), std::forward<Mat2>(m2));
 ```
+// Require the scalar type is an std::vector
+template <typename T>   
+require_st_std_vector = require_t<is_std_vector<scalar_type_t<std::decay_t<T>>>>;
 
-#### Adding a new requires
+// Ex: Used to define a signature for `std::vectors` with a scalar type that is autodiffable
+template <typename StdVec, require_std_vector_st<is_var, StdVec>* = nullptr>
+auto my_func(StdVec&& vec);
+```
 
-For a full list of predefined `requires` template parameters please see @ref require_meta.
+In the below, `{TYPE_TRAIT}` represents the name of the trait the
+requires checks. Each new require _must_ follow this standard API.
 
-If you are adding a new type trait that contains a `bool` member named value, you can add it to the set of known `requires` template parameters by using the macros defined in @ref stan/math/prim/meta/require_helpers.hpp .
+1. `require_{TYPE_TRAIT}_t`: The template parameter must return `true`
+when passed to the type trait
 
-For an example, we will use the `is_double_only` type trait below that only has a `bool value` member equal to `true` if the type `T` is `double`.
+2. `require_not_{TYPE_TRAIT}_t`: The template parameter must return
+`false` when passed to the type trait
 
-```cpp
-template <typename T>
-struct is_double_only
-    : bool_constant<std::is_same<double, std::decay_t<T>>::value> {};
+3. `require_all_{TYPE_TRAIT}_t`: The template parameters must all return
+`true` when passed to the type trait
 
-```
+4. `require_all_not_{TYPE_TRAIT}_t`: The template parameters must all
+return `false` when passed to the type trait
 
-We can add the associated `requires` template parameters for the standard requires of `require_double_only_t` using the @ref STAN_ADD_REQUIRE_UNARY macro.
-We supply the name to use in the require (`double_only`), The type trait name (`is_double_only`) and which group this type trait is in `require_stan_scalar_real`.
+5. `require_any_{TYPE_TRAIT}_t`: At least one of the template parameters
+must return `true` when passed to the type trait
+
+6. `require_any_not_{TYPE_TRAIT}_t`: At least one of the template
+parameters must return `false` when passed to the type trait
+
+In addition to all the requires with an `_t` at the end, the requires
+also have `_st`, `_vt` variants where in addition to the logic above,
+the @ref value_type or @ref scalar_type must follow the same logic as
+the type for `T`. The `_st_`, and `_vt_` variants must also follow the
+same logic but for checking only the inner @ref value_type or
+@scalar_type.
```

Comment:
This is all good! Should we add a link somewhere near the top for the docs for all the currently available requires? 

https://mc-stan.org/math/group__require__meta.html

Reply:
Yes, I'll look for a good place for this.

Reply:
Ah. I did link it! (using the `@ref require_meta` tag)

L69 in this doc. It looks like:
```
For convenience in using this technique, the Math library has
implemented a set of [`requires` type traits](@ref require_meta).
```

I'll look for a second place to drop the link.